### DISCUSSION TITLE
Could not load file or assembly `System.Configuration.ConfigurationManager`

### DISCUSSION THREAD

**Discussion by unknownRILL** [nxrighthere/UnrealCLR#155] [Help] [answered]
Hi,

I added a dependency .dll that was intended for .Net Framework 4.5 or later but when I try to call something from it I get this error in ue4: 

> LogUnrealCLR: Error: UnrealCLR::Module::Exception: System.IO.FileNotFoundException: Could not load file or assembly 'System.Configuration.ConfigurationManager, Version=0.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The system cannot find the file specified.
> File name: 'System.Configuration.ConfigurationManager, Version=0.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'

from googling around it seems like it has to do with compatibility issues between .net core and the .dll I'm using and System.Configuration not being compatible with core?

I tried using NuGet to install System.Configuration but that doesn't change the error and I also tried to NuGet https://www.nuget.org/packages/Microsoft.Windows.Compatibility
However that results in a similar error:

> 
> LogUnrealCLR: Error: UnrealCLR::Module::Exception: System.IO.FileNotFoundException: Could not load file or assembly 'System.Configuration.ConfigurationManager, Version=0.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The system cannot find the file specified.
> File name: 'System.Configuration.ConfigurationManager, Version=0.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
>    at TheDLL.Net.Bluetooth.BluetoothFactoryConfig.get_KnownStacks()
>    at TheDLL.Net.Bluetooth.Factory.BluetoothFactory.GetStacks_inLock()
>    at TheDLL.Net.Bluetooth.Factory.BluetoothFactory.get_Factories()
>    at TheDLL.Net.Bluetooth.BluetoothRadio.get_AllRadios()
>    at TheDLL.Devices.Common.DeviceManager`1.get_IsBluetoothRadioEnabled()
>    at UnrealEngine.Tests.StaticMeshes.OnBeginPlay() in C:\Users\MyName\Downloads\UnrealCLR-1.29\Source\Managed\Tests\StaticMeshes.cs:line 39
>    at UnrealEngine.Tests.Main.OnWorldPostBegin() in C:\Users\MyName\Downloads\UnrealCLR-1.29\Source\Managed\Tests\Tests.cs:line 42
>    at UnrealEngine.Runtime.Core.ManagedCommand(Command command) in C:\Users\MyName\Downloads\UnrealCLR-1.29\Source\Managed\Runtime\Runtime.cs:line 150
> 

I even tried to 

` <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies> `

but I still get the same error.
I'm still rather new to .net and vs2019 if anyone could shed some light on how I can get UnrealCLR to be compatible with System.Configuration or if it is and I'm just not setting it up right please tell me how to properly set it up. If this is just something that's not going to be possible I'd also like to know since I do not have the source code for the .dll I'm trying to use so I can't change the parts that are using System.Configuration.

Thank you!

**Comment by nxrighthere** [comment]
Do you [publish](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-publish) a project or just build it?

**Reply by nxrighthere** [reply]
Related https://github.com/nxrighthere/UnrealCLR/discussions/108 and https://github.com/nxrighthere/UnrealCLR/discussions/71.

**Reply by unknownRILL** [reply]
Hi, I was originally just building, however I have now tried it using `dotnet publish -f net5.0 -c Release` I see that it generates a publish folder that has the .dll and also a bunch of new system .dlls however when I copy the contents of the publish folder into the correct folder in ue4 I'm still getting the same error. I have tried publishing as release, debug, with the Compatibility package, then the Configuration package, then both.

So far no luck however I did notice something that might be related:

> LogUnrealCLR: Error: UnrealCLR::Module::Exception: System.IO.FileNotFoundException: Could not load file or assembly 'System.Configuration.ConfigurationManager, Version=0.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The system cannot find the file specified.
> File name: 'System.Configuration.ConfigurationManager, Version=0.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
>    at Namespace2.Net.Bluetooth.BluetoothFactoryConfig.get_KnownStacks()
>    at Namespace2.Net.Bluetooth.Factory.BluetoothFactory.GetStacks_inLock()
>    at Namespace2.Net.Bluetooth.Factory.BluetoothFactory.get_Factories()
>    at Namespace2.Net.Bluetooth.BluetoothRadio.get_AllRadios()
>    at UnrealEngine.Tests.StaticMeshes.OnBeginPlay() in C:\Users\MyName\Downloads\UnrealCLR-1.29\Source\Managed\Tests\StaticMeshes.cs:line 40
>    at UnrealEngine.Tests.Main.OnWorldPostBegin() in C:\Users\MyName\Downloads\UnrealCLR-1.29\Source\Managed\Tests\Tests.cs:line 43
>    at UnrealEngine.Runtime.Core.ManagedCommand(Command command) in C:\Users\MyName\Downloads\UnrealCLR-1.29\Source\Managed\Runtime\Runtime.cs:line 150

Previously it also included between the first Namespace2 line and the atUnrealEngine.Tests.StaticMeshes line this other line that is now missing:

> Namespace1.Devices.Common.DeviceManager`1.get_IsBluetoothRadioEnabled()

The .dll contains two namespaces and it seems this first namespace that includes the actual call I'm making is no longer throwing an error. Not sure if it is related or not but seems like I'm getting closer?

Thanks again for all your help!

**Reply by nxrighthere** [reply]
Could you please send me a minimal project to reproduce this?

**Reply by unknownRILL** [reply]
I've shared the 7zipped folder with you via Dropbox through your gmail. I just used the built in Test csproj. the DLL I'm trying to get working is in the THE_DLL folder in root. The only changes I made are in the staticmeshes.cs file for the Test csproj. When I run the tests level with the test systems set to staticmeshes then I see the errors mentioned above. the dll folder also includes a link to the webpage where you can download the dll and see the api. The published dlls should be in:

> \UnrealCLR-1.29_Modified_With_Errors\Source\Managed\Tests\bin\Release\win10-x64\publish

I don't know if what I'm trying to have it do is just not feasible or if I'm just going about it wrong but if you could take a look I would greatly appreciate it, thank you!

**Comment by nxrighthere** [comment, marked as answer]
The way you try to use the Bio Radio SDK library will not work:

- It's a .NET Framework 4.5 library and it will not work with CoreCLR
- Referencing and copying dependencies from NuGet separately will not help, all references should be specified in the Bio Radio SDK library project and published from there

**Reply by unknownRILL** [reply]
Got it thanks for your time, I just wanted to make sure this was the case and not just because I had missed something about the plugin.
